bazel/protobuf.patch

```diff
@@ -137,13 +137,14 @@ index 37d80e591..3dddd4ecf 100644
  // This error has been generally flaky, but we need to disable it specifically
  // to fix https://github.com/protocolbuffers/protobuf/issues/12313
  #pragma clang diagnostic ignored "-Wunused-parameter"
-@@ -925,6 +925,9 @@ static_assert(PROTOBUF_ABSL_MIN(20230125, 3),
+@@ -925,7 +925,9 @@ static_assert(PROTOBUF_ABSL_MIN(20230125, 3),
  #pragma warning(disable: 4125)
  #endif
  
-+#if defined(__GNUC__)
-+#pragma GCC diagnostic ignored "-Wundef"
-+#endif
- #if PROTOBUF_ENABLE_DEBUG_LOGGING_MAY_LEAK_PII
+-#if PROTOBUF_ENABLE_DEBUG_LOGGING_MAY_LEAK_PII
++#ifndef PROTOBUF_ENABLE_DEBUG_LOGGING_MAY_LEAK_PII
```

Comment:
I suppose you could rewrite this as something like:

```
#if defined(PROTOBUF_ENABLE_DEBUG_LOGGING_MAY_LEAK_PII) && PROTOBUF_ENABLE_DEBUG_LOGGING_MAY_LEAK_PII
#define PROTOBUF_DEBUG true
#else
#define PROTOBUF_DEBUG false
#if
```

Would that work?

Reply:
I could, it would just lead to a very long line. I've also sent this upstream and if they take it with a change I'll further tweak it here.

Reply:
Sounds good to me. LMK when this is ready for review (and covert it to non-draft) and I'll LGTM!

Reply:
We do have `\` line continuation for a reason, I suppose, and this is less code duplication, which is a certainly a win.